--- a/mdns_tinysvcmdns.c	2026-09-26 01:42:47.898514732 +0200
+++ b/mdns_tinysvcmdns.c	2026-09-26 01:42:47.906514784 +0200
@@ -69,6 +69,11 @@
   gethostname(hostname, 99);
   // according to POSIX, this may be truncated without a final NULL !
   hostname[99] = 0;
+  // Android always reports "localhost", which would clash on the network.
+  if (strcmp(hostname, "localhost") == 0)
+    snprintf(hostname, 99, "shairport-%02X%02X%02X%02X%02X%02X", config.hw_addr[0],
+             config.hw_addr[1], config.hw_addr[2], config.hw_addr[3], config.hw_addr[4],
+             config.hw_addr[5]);
 
   // will not work if the hostname doesn't end in .local
   char *hostend = hostname + strlen(hostname);
